src/main/java/io/vertx/core/http/impl/ConnectionManager.java

```diff
@@ -281,8 +294,14 @@ private void createNewConnection(Waiter waiter) {
      */
     Waiter getNextWaiter() {
       Waiter waiter = waiters.poll();
+      if (waiter != null && ConnectionManager.this.metrics.isEnabled()) {
+        ConnectionManager.this.metrics.dequeueRequest(metric, waiter.metric);
```

Comment:
Thinking out loud.

The SPI implementer has no way to determine if `dequeue` is called for a canceled metric? Maybe we don't care because we want to measure waiting time in the queue for both waiters who passed and who timed out.